lib/tlIO/OpenEXRRead.cpp

```diff
@@ -153,76 +154,87 @@ namespace tl
+                        if (auto logSystem = logSystemWeak.lock())
                         {
-                        case Imf::PixelType::HALF:
-                            info.pixelType = imaging::getFloatType(layer.channels.size(), 16);
-                            break;
-                        case Imf::PixelType::FLOAT:
-                            info.pixelType = imaging::getFloatType(layer.channels.size(), 32);
-                            break;
-                        case Imf::PixelType::UINT:
-                            info.pixelType = imaging::getIntType(layer.channels.size(), 32);
-                            break;
-                        default: break;
+                            const std::string id = string::Format("tl::io::exr::Read {0}").arg(this);
+                            std::vector<std::string> s;
+                            s.push_back(string::Format(
+                                            "\n"
+                                            "    file name: {0}\n"
+                                            "    display window {1}\n"
+                                            "    data window: {2}\n"
+                                            "    compression: {3}").
+                                        arg(fileName).
+                                        arg(_displayWindow).
+                                        arg(_dataWindow).
+                                        arg(getLabel(header.compression())));
+                            const auto& channels = header.channels();
+                            for (auto i = channels.begin(); i != channels.end(); ++i)
+                            {
+                                std::stringstream ss2;
+                                ss2 << "    channel " << i.name() << ": " << getLabel(i.channel().type) << ", " << i.channel().xSampling << "x" << i.channel().ySampling;
+                                s.push_back(ss2.str());
+                            }
+                            logSystem->print(id, string::join(s, '\n'));
                         }
-                        if (imaging::PixelType::None == info.pixelType)
+
+                        // Get the tags.
+                        readTags(header, _info.tags);
+
+                        // Get the layers.
+                        
+                        std::vector<Layer> layers = getLayers(header.channels(), channelGrouping);
+                        size_t offset = _info.video.size();
+                        _info.video.resize( offset + layers.size() );
+                        for (size_t i = 0; i < layers.size(); ++i)
                         {
-                            throw std::runtime_error(string::Format("{0}: Unsupported image type").arg(fileName));
+                            layers[i].partNumber = partNumber;
+                            const auto& layer = layers[i];
+                            _layers.push_back( layer );
```

Comment:
Could you make a pass over the changes to make the formatting consistent? (e.g., no spaces between parentheses)